Multi onesky projects for translations

BourotBenjamin opened the pull request:
Hi, 

I updated this project because we some specific needs.
We have one Symphony project, and we need to send translations in more than one OneSky's projects.
So here is the solution I provide : 
I updated the config like that : 

    file_paths:
        123:
            - "%kernel.root_dir%/Resources/translations"
        456:
            - "%kernel.root_dir%/Resources/entities-translations"

With this modifications, we set a project id for one (or more) paths and it upload files directly on good projects.

In addition, I add to the commands a optional "projectId" for the commands (which is needed when you set filePath option ) : 
If you set only filePath, as he don't know his project id, he pull or push every files from every projects
If you only set projectId, he pulls (or push) every files from (to) the wanted project.
If you set filePath and projectId, he pulls (or push) files to (from) folders given from (to) the wanted project.

romainkuzniak commented:
Thank you for this PR!

I think a better way to handle the different projects should be to enable an array of project in the config and default settings.
This way, each project could override the default settings

So the config file should look like this:
``` yaml
# app/config/config.yml

openclassrooms_onesky:
    api_key:  %onesky.api_key%
    api_secret: %onesky.api_secret%
    source_locale: %source_locale% #optional, default en
    locales:
        - fr
        - es
    file_format: %onesky.file_format% #optional, default xliff
    keep_all_strings: false # default true
    projects:
        projectName:
            id: %onesky.project_id%
            source_locale: %source_locale% #optional, default main setting
            locales: #optional, default main setting
                - fr
                - es
            file_format: %onesky.file_format% #optional, default main setting
            file_paths:
                - %path.to.translations.files.directory%
            keep_all_strings: true #optional, default main setting

```
You have to know that file path could be wild card. I’m going to update the current documentation.
Another thing, you PR broke the build, and decrease the cover. Could you please fix it?

Thank you

BourotBenjamin commented:
I saw, I would try to fix it tomorrow. ;)

BourotBenjamin commented:
I fixed the tests to work with the new config, I also had to change the guzzle exception as this exception doesn't' event appears on Guzzle git repo.
It seems to be an Amazon AWS exclusive exception. 
So I changed it for a Symphony's HTTP Exception with status code = 500.

BourotBenjamin commented:
Hi,

I updated this PR but I didn't manage to find how to handle two different travis configurations for the tests.
As php 7 needs yaml-2.0.0 and php 5.* needs yaml pecl extensions, php 5.* build would failed so I removed it from the travis configuration.
But if you new how to handle this case, I would be happy to know. :)

Thanks.
